tests/integration/test_charm.py

```diff
@@ -0,0 +1,63 @@
+import logging
+from pathlib import Path
+
+import pytest
+import yaml
+from pytest_operator.plugin import OpsTest
+import lightkube
+from lightkube.generic_resource import create_global_resource
+
+log = logging.getLogger(__name__)
+
+METADATA = yaml.safe_load(Path("./metadata.yaml").read_text())
+
+APP_NAME = "kfserving"
+CHARM_ROOT = "."
+
+
+@pytest.mark.abort_on_fail
+async def test_build_and_deploy(ops_test):
+    # build charm
+    charm_under_test = await ops_test.build_charm(CHARM_ROOT)
+    log.info(f"Built charm {charm_under_test}")
+
+    # deploy charm
+    image_path = METADATA["resources"]["oci-image"]["upstream-source"]
+    resources = {"oci-image": image_path}
+    await ops_test.model.deploy(charm_under_test, resources=resources)
+
+    # wait until active and idle
+    await ops_test.model.wait_for_idle(
+        apps=[APP_NAME], status="active", raise_on_blocked=True, timeout=60 * 10
+    )
+
+    # check status
+    charm_status = ops_test.model.applications[APP_NAME].units[0].workload_status
+    log.info(f"charm {charm_under_test} status is {charm_status}")
+    assert charm_status == "active"
+
+
+async def test_flower_predictor(ops_test: OpsTest, lightkube_client):
+    log.info("ERROR: Test not yet implemented")
+    # TODO: implement this test once kfserving charm and knative work properly
+    # test to reproduce with pytest:
+    # set -eux
+    # kubectl apply -nci-test -f examples/flower.yaml
+    # sleep 10
+    # kubectl wait -nci-test --for=condition=available deployment --all --timeout=5m
+    # ADDRESS=$(kubectl get pods -nci-test -lcomponent=predictor -oyaml \
+    #   | yq e .items.0.status.podIP -)
+    # curl -v \
+    #   -d @examples/flower_input.json \
+    #   -H "Content-Type: application/json" \
+    #   http://$ADDRESS:8080/v1/models/my-model:predict
+
+
+# Helpers
+@pytest.fixture(scope="session")
+def lightkube_client() -> lightkube.Client:
+    client = lightkube.Client()
+    create_global_resource(
```

Comment:
Does this actually do anything? I thought this function just created a `Profile` object that you could then pass to methods such as `client.get`. AFAIK it doesn't do anything like talk to the Kubernetes API to create the CRD or anything.

Reply:
Hi Ken! This is in the code since `test_flower_predictor_test` requires a `lightkube_client`. Since `test_flower_predictor_test` doesn't execute any code it isn't being used. 

The idea was to use it as the [kubeflow-profile-operator ](https://github.com/canonical/kubeflow-profiles-operator/blob/9d6e97515ffa7af227aa7bce4c38b66d1f4dcf3c/tests/integration/test_charm.py#L53-L61) does. From what I can tell in the kubeflow-prfoiles-operator it can be used to delete/create/inquire within kubernetes cluster. Due to the issues with knative & kfserving versions the method `test_flower_predictor` wasn't written and hence you don't see anything along the lines of `client.get`,`client.delete`, etc.

When the issues with knative & kfserving versions get resolved I believe this will be needed within the code. Let me know if you think I should remove it.

Reply:
ping @ca-scribner on this. AFAICT just calling `create_global_resource` won't do anything useful?

Reply:
`create_global_resource` is needed when you have a CRD that you want lightkube to create instances of.  In the case of the profiles operator, we needed to create `Profile` objects so we first run `create_global_resource(...profile stuff...)`, and that **registers** the `Profile` CRD within lightkube.  iirc it does return something related to that registration, but you don't need it to create any CRs from the new definition.

In this operator's test code, @MiaAltieri if you need to create kubeflow's `Profile` object then it is great as is.  But if you wanted to create some CRs provided by kfserving/seldon, etc, you'd need to use the appropriate settings (`group=myGroup, version=myVersion, kind=myKind`).  In either case, if you're blocked from using it (eg: something is blocking you from doing `test_flower_predictor`) I'd vote to remove this code for now and just bring it in once the other issues are resolved (that way there's no unused code blocks left around).

Reply:
Thanks for the clear explanation! This all makes sense now. I'll remove it right now.